isis/src/base/apps/trackextract/trackextract.cpp

```diff
@@ -0,0 +1,221 @@
+    msg += " If you want to create a tracking cube, run a mosaic program.";
+    throw IException(IException::Programmer, msg, _FILEINFO_);
+  }
+}
+
+
+/**
+ * Creates the mosaic cube by copying the input cube without the tracking band.
+ * Then, removes the tracking table from the mosaic label and adds a group pointing
+ * to the tracking cube.
+ *
+ * @param inputName   The name of the input cube
+ * @param ouputName   The name of the output cube
+ * @param bandsVector The indices of the bands that are not the tracking band
+ */
+void createMosaicCube(QString inputName, QString outputName, QVector<QString> bandsVector) {
+  ProcessByLine p;
+
+  CubeAttributeInput inAtt = CubeAttributeInput();
+  inAtt.setBands(bandsVector.toStdVector());
+  
+  p.SetInputCube(inputName, inAtt);
+  p.SetOutputCube("TO");
+  p.StartProcess(copyPixels);
+  p.EndProcess();
+
+  Cube mosaicCube;
+  try {
+    mosaicCube.open(outputName,"rw");
+  }
+  catch (IException &e) {
+    throw IException(IException::User,
+                     "Unable to open the file [" + outputName + "] as a cube.",
+                     _FILEINFO_);
+  }
+
+  if (!mosaicCube.deleteBlob("Table", "InputImages")) {
+    QString msg = "The input cube [" + inputName + "] does not have a tracking table.";
+    throw IException(IException::Programmer, msg, _FILEINFO_);
+  }
+
+  // Add Tracking Group to the mosaic cube
+  PvlGroup trackingGroup = PvlGroup("Tracking");
+  PvlKeyword trackingName = PvlKeyword("Filename");
+  FileName cubeName = FileName(outputName);
+  trackingName.setValue(cubeName.baseName() + "_tracking.cub"); //Strip off path and add _tracking
+  trackingGroup.addKeyword(trackingName);
+  mosaicCube.putGroup(trackingGroup);
+
+  mosaicCube.close();
+}
+
+
+/**
+ * Creates the tracking cube by copying the input cube with only the tracking band.
+ * Then, goes through each pixel and subtracts the input cube's pixel type's min value
+ * because ProcesMosaic used to add the min value to each pixel.
+ *
+ * @param inputName    The name of the input cube
+ * @param ouputName    The name of the output cube
+ * @param trackingBand The index of the tracking band
+ */
+void createTrackCube(QString inputName, QString ouputName, int trackBand) {
+  ProcessByLine p;
+
+  CubeAttributeInput inAtt = CubeAttributeInput("+" + QString::number(trackBand));
+  p.SetInputCube(inputName, inAtt);
+
+  FileName cubeName = FileName(ouputName);
+  // Strip off any extensions and add _tracking
+  QString trackingName = cubeName.path() + "/" + cubeName.baseName() + "_tracking.cub";
+  Cube inputCube = Cube(inputName);
+  int numSample = inputCube.sampleCount();
+  int numLine = inputCube.lineCount();
+
+  CubeAttributeOutput outAtt;
+  outAtt.setPixelType(UnsignedInteger);
+  outAtt.setMinimum(VALID_MINUI4);
+  outAtt.setMaximum(VALID_MAXUI4);
+
+  p.SetOutputCube(trackingName, outAtt, numSample, numLine);
+
+  p.StartProcess(copyTrackPixels);
+  p.EndProcess();
```

Comment:
Is the new tracking table ever written to the tracking cube? This app needs to write an updated tracking table to the tracking cube.

Reply:
I'm assuming that the existing table from the input cube is copied to the new tracking cube by ProcessByLine, but it'd be nice if it was in the new format with the indices as a 3rd field. You could initialize a TrackingTable object from the input cube's table and then immediately call TrackingTable::toTable and write that to the tracking cube.